DQM/HGCal/plugins/HGCalTriggerClient.cc

```diff
@@ -84,103 +204,434 @@ void HGCalTriggerClient::analyze(const edm::Event& iEvent, const edm::EventSetup
 
   // fill histogram for trigtime distribution
   const auto& metadata = iEvent.get(metadataToken_);
-  int trigTime = metadata.trigTime_;
-  h_trigtime->Fill(trigTime);
+  uint32_t trigTime = metadata.trigTime_;
+  h_trigtime->Fill(int(trigTime));
+
+  uint32_t trigType = metadata.trigType_;
 
   // read ECON-T raw data
   const auto& raw_trig_data = iEvent.get(trigRawToken_);
   const auto& data = raw_trig_data.FEDData(0);
 
+  // const auto recEvent = iEvent.get(recRun_);
+  // const hgcal_slinkfromraw::RecordRunning *recRecord =  recEvent->getTrigRecRun();
+
   // vector<char> ==> uint64_t*
   const uint64_t* payload = (uint64_t*)(data.data());
   const size_t payload_size = data.size() / sizeof(uint64_t);
 
   edm::LogSystem out("HGCalTriggerClient");
-  if (debug_) {
-    out << std::hex << std::setfill('0');
-    for (unsigned i = 0; i < payload_size; i++) {
-      out << std::setw(4) << i << "  " << std::setw(16) << payload[i] << '\n';
+  if (debug_)
+    eventDump(payload, payload_size, out, 5);
+
+  // ====== ECON-T unpacker ======
+
+  bool isBxE0 = false;
+  bool isBxE1 = false;
+  // bool isBxCE0 = false;
+  // bool isBxCE1 = false;
+  bool isSTCNE0 = false;
+  bool isSTCNE1 = false;
+  bool isSTCLE0 = false;
+  bool isSTCLE1 = false;
+  bool isEngE0 = false;
+  bool isEngE1 = false;
+
+  if (trigType == 0x0001)
```

Comment:
This should be a `switch` and `trigType` an instance of an `enum` so that you can benefit from [compile-time warnings and optimisations](https://brevzin.github.io/c++/2019/08/01/enums-default/).

Reply:
Necessary modifications have been applied (see [8165d07](https://github.com/CMS-HGCAL/cmssw/pull/86/commits/8165d077af54e5ab2fcceb6c538b92669bbe113d))